library/manageiq.py

```diff
@@ -180,13 +191,43 @@ def generate_endpoints(self, hostname, port, token, h_hostname, h_port):
                                                      'auth_key': token}})
         return endpoints
 
+    def delete_provider(self, provider_name):
+        """ Deletes the provider
+
+        Returns:
+            the delete task id if a task was generated, whether or not
+            a change took place and a short message describing the operation
+            executed
+        """
+        message = ""
+        task_id = None
+        provider_id = self.find_provider_by_name(provider_name)
+        if provider_id:
+            try:
+                result = self.client.post('{providers_url}/{id}'.format(providers_url=self.providers_url, id=provider_id),
+                                          action='delete')
+                if result['success']:
+                    message = result['message']
+                    task_id = result['task_id']
+                    self.changed = True
```

Comment:
Do you need `self.changed` instance var to outlive this method?
I also don't like the long flow `message = ""; task_id = None` — setting them here — `res_args =...; return res_args`.

I'd immediately `return dict(task_id=result['task_id'], changed=True, msg=result['message'])` here,
and separate `return dict(changed=False, msg="... doesn't exist"...)` at the end.